doc/wrapping_js_components.md

```diff
@@ -0,0 +1,693 @@
+
+### Theoretical Conclusion
+
+This section gave a high level view of how wrapping a JS component works in theory. You must wrap the JS component in such a way that it follows OverReact's pattern of using `UiFactory` to create a props instance capable of building a `ReactElement`. This is possible with `uiJsComponent`, which takes in a `ReactJsComponentFactoryProxy` that points to a raw JS component attached to the `Window`.
+
+The rest of this guide will reference these same concepts as it implements the code step-by-step.
+
+### Implementation
+
+With the theory in mind, this section implements the necessary pieces step by step.
+
+### Pre-Requisites
+
+Before jumping into wrapping your component, here is a list of things to do to get set up:
+
+- Make sure your package depends on `over_react: ^4.4.0`.
+- Have a JS bundle and load the JS script in any applications that will use the component (in an `example` directory, test template, `app` directory, etc). **NOTE** that if your package is consumed by a different one, it is important to ensure that the consuming package(s) know to load the script in their applications too.
+- Create a file for your component. The file name does not have to be unique for this process and can be consistent with existing naming patterns in your package.
+
+### Adding the JS Variable
+
+The first step is to create the interop getter to make the JS implementation accessible in Dart. The code written for this section is shown below.
+
+```dart
+@JS()
+library some_library.path.to.component;
+
+import 'package:js/js.dart';
+
+@JS('MagicalJsPackage.ArbitraryComponent')
+external ReactClass get _jsArbitraryComponent;
+```
+
+There are three pieces here:
+
+1. Adding directives
+1. Creating the getter
+1. Attaching the annotation
+
+#### Adding the Directives
+
+To get started, add the important directives to the top of the file.
+
+```dart
+@JS()
+library some_library.path.to.component;
+
+import 'package:js/js.dart';
+```
+
+The `js.dart` import gives access to the `JS()` annotation. The `JS()` annotation is necessary because the file is defining interop APIs. Lastly, the `library` directive is necessary because the `JS()` annotation must be attached to a specific library.
+
+#### Creating the Getter
+
+Creating the getter is very simple.
+
+```dart
+external ReactClass get _jsArbitraryComponent;
+```
+
+The code here can basically be copied and pasted from the code found in this doc, just changing the name of the getter to be something more specific than `_jsArbitraryComponent`. However, the name can really be anything. We have the norm of using `_js{ComponentName}` because:
```

Comment:
#nit In the context of stylistic programming guidelines, "convention" would be more standard way to refer to this than "norm".
```suggestion
The code here can basically be copied and pasted from the code found in this doc, just changing the name of the getter to be something more specific than `_jsArbitraryComponent`. However, the name can really be anything. We have the convention of using `_js{ComponentName}` because:
```